tensorflow/python/ops/metrics_impl.py

```diff
@@ -2633,11 +2633,10 @@ def root_mean_squared_error(labels,
   predictions, labels, weights = _remove_squeezable_dimensions(
       predictions=predictions, labels=labels, weights=weights)
   mse, update_mse_op = mean_squared_error(labels, predictions, weights, None,
-                                          None, name or
-                                          'root_mean_squared_error')
+                                          None, name='mean_squared_error')
 
-  rmse = math_ops.sqrt(mse)
-  update_rmse_op = math_ops.sqrt(update_mse_op)
+  rmse = math_ops.sqrt(mse, name=name or 'root_mean_squared_error')
```

Comment:
I think you either have to put parens around the expression or assign 

name or 'root_mean_squared_error'

to a temp variable.

Reply:
I don't think this is the issue, if you look at the original lines 2636 and 2637, they had a similar expression with no parens or temp variable.